src/qiboml/visualization/visualization.py

```diff
@@ -0,0 +1,171 @@
+from mpl_toolkits.mplot3d import Axes3D
+import matplotlib.pyplot as plt
+import numpy as np
+from qibo import hamiltonians
+from qibo.symbols import X, Y, Z
+
+
+class Bloch():
+    def __init__(self):
+        self.fig = plt.figure()
+        self.ax = self.fig.add_subplot(111, projection='3d', elev=30)
+
+    def save_plot(self, name):
+        "Function to save the sphere."
+        self.ax.set_aspect("equal")
+        plt.axis('off')
+        plt.tight_layout()
+        plt.savefig(name)
+
+    def plot(self):
+        "Function to plot the sphere."
+        self.ax.set_aspect("equal")
+        plt.axis('off')
+        plt.tight_layout()
+        plt.show()
+
+    
+    def label_states(self, nclasses):
```

Comment:
This function should be very problem dependent, right?
If your idea is to encode a 3d polygon inside the sphere, then I would suggest to just define a method `bs.distribute_labels_on_surface` or `bs.inscribed_solid(n_vertexes=10)`. This will be very useful to show the inscribed solid and also, if needed, to return the label states corresponding to the vertexes.